src/TLuaInterpreter.cpp

```diff
@@ -17158,49 +17162,115 @@ void TLuaInterpreter::initIndenterGlobals()
-                  .arg(QCoreApplication::applicationDirPath())
-                  .toUtf8().constData());
+#if defined(Q_OS_UNIX)
+    // Now also included for the MacOs case so that MacOS developers can build
+    // from within the Qt Creator IDE
+
+    /*
+     * Additional paths for the lua package search path - ? is the "package"
+     * name though in the case for the lua code formatter it itself includes
+     * leading path elements of "lcf" and "workshop" in the
+     * require("lcf.workshop.base") case - the lua script being sought is the
+     * "base.lua" file but our paths must give the loction of the "lcf"
+     * directory two parent directories up:
+     * 1 installed *nix case - probably not applicable to Windows
+     *     "LUA_DEFAULT_PATH/?.lua" (if defined and not empty)
+     * 2 AppImage (directory of executable):
+     *     "<applicationDirectory>/?.lua"
+     * 3 QMake shadow builds without CONFIG containing "debug_and_release" but
+     *   with "debug_and_release_target" (default on most OS but NOT Windows):
+     *     "<applicationDirectory>/../3rdparty/?.lua"
+     * 4 QMake shadow builds with CONFIG containing "debug_and_release" AND
+     *   "debug_and_release_target" (usually Windows):
+     *     "<applicationDirectory>/../../3rdparty/?.lua"
+     * 5 CMake shadow builds
+     *     "<applicationDirectory>/../../mudlet/3rdparty/?.lua"
+     * NOTE: In the cmake case the shadow build directory is alongside the base
+     * directory of the extracted source package so it MUST be extracted into
+     * a known directory name which is assumed here to be "mudlet"!
+     */
 
-    // if using LuaJIT, adjust the cpath to look in /usr/lib as well - it doesn't by default
-    luaL_dostring(pIndenterState.get(), "if jit then package.cpath = package.cpath .. ';/usr/lib/lua/5.1/?.so;/usr/lib/x86_64-linux-gnu/lua/5.1/?.so' end");
+    if (!QStringLiteral(LUA_DEFAULT_PATH).isEmpty()) {
+        luaL_dostring(pIndenterState.get(), QStringLiteral("package.path = [["
+                                                           LUA_DEFAULT_PATH "/?.lua;" // 1
+                                                           "%1/?.lua;" // 2
+                                                           "%1/../3rdparty/?.lua;" // 3
+                                                           "%1/../../3rdparty/?.lua;" // 4
+                                                           "%1/../../mudlet/3rdparty/?.lua;]]" // 5
+                                                           " .. package.path")
+                      .arg(QCoreApplication::applicationDirPath()).toUtf8().constData());
+    } else {
+        luaL_dostring(pIndenterState.get(), QStringLiteral("package.path = [["
+                                                           "%1/?.lua;" // 2
```

Comment:
all of this section is reproduced code. Instead of two big repeated sections, why not do the push of the LUA_DEFAULT_PATH in the if, and then push the rest of them outside of the if/else and also keep it DRY.

If the existing package.path absolutely must be behind all of it, then create the string first and then adjust package.path. But this huge repeated block is not only an eyesore, it's just begging for someone to miss an update in one or the other of them.

Reply:
Okay I've tried a different approach - is that better now?